Python/pywarpx/diagnostics/data/BaseReader.py

```diff
@@ -0,0 +1,189 @@
+import collections
+import os
+import re
+
+import numpy as np
+import pandas as pd
+
+
+class DataReader():
+    """
+    Base class that all data readers should inherit from.
+    Works only in Cartesian geometry.
+    """
+
+    def __init__(self, run_directory, file_prefix=None):
+        """
+        Parameters
+        ----------
+        run_directory: string
+            Path to the run directory of WarpX.
+        """
+        if run_directory is None:
+            raise ValueError('The run_directory parameter can not be None!')
+
+        self.run_directory = run_directory
+        self.data_file_prefix = None
+        self.data_file_suffix = None
+
+        self.has_species = None
+
+    def get_data_path(self, subdir='reducedfiles'):
+        """
+        Returns the path to the data file.
+        Parameters
+        ----------
+        subdir: string
+            Name of reducedfiles directory, default is 'reducedfiles'.
+        Returns
+        -------
+        A string with a path to the current diagnostic (assumed unique).
+        """
+
+        sim_output_dir = os.path.join(self.run_directory, 'diags/'+subdir+'/')
+        if not os.path.isdir(sim_output_dir):
+            raise IOError('The simulation directory does not exist inside '
+                          'path:\n  {}\n'
+                          'Did you set the proper path to the run directory?\n'
+                          'Did the simulation already run?'
+                          .format(self.run_directory))
+
+        data_file_path = os.path.join(
+            sim_output_dir,
+            self.data_file_prefix +
+            self.data_file_suffix
+        )
+        if not os.path.isfile(data_file_path):
+            raise IOError('The file {} does not exist.\n'
+                          'Did the simulation already run?'
+                          .format(data_file_path))
```

Comment:
This is purely a preference thing, but I like to use f-strings nowadays, since they are more readable in my opinion:
```suggestion
            raise IOError(f'The file {data_file_path} does not exist.\n'
                          'Did the simulation already run?'
                          )
```